core/src/test/java/org/bitcoinj/uri/BitcoinURITest.java

```diff
@@ -85,8 +86,13 @@ public void testConvertToBitcoinURI() {
         NetworkParameters alternativeParameters = new MockAltNetworkParams();
         String mockNetGoodAddress = MockAltNetworkParams.MOCKNET_GOOD_ADDRESS;
 
-        assertEquals("mockcoin:" + mockNetGoodAddress + "?amount=12.34&label=Hello&message=AMessage",
-             BitcoinURI.convertToBitcoinURI(LegacyAddress.fromBase58(alternativeParameters, mockNetGoodAddress), parseCoin("12.34"), "Hello", "AMessage"));
+        Networks.register(alternativeParameters);
+        try {
+            assertEquals("mockcoin:" + mockNetGoodAddress + "?amount=12.34&label=Hello&message=AMessage",
+                    BitcoinURI.convertToBitcoinURI(LegacyAddress.fromBase58(alternativeParameters.network(), mockNetGoodAddress), parseCoin("12.34"), "Hello", "AMessage"));
+        } finally {
+            Networks.unregister(alternativeParameters);
+        }
```

Comment:
I sense a deja-vu here (-:

Probably also a candidate for separation.

Reply:
Yeah, this can be done separately, although the register/unregister is not necessary when `LegacyAddress` is storing `NetworkParameters`.